runner/ollamarunner/cache.go

```diff
@@ -267,7 +275,22 @@ func (c *InputCache) ShiftCacheSlot(slot *InputCacheSlot, numKeep int32) error {
 	if c.cache != nil {
 		err := c.cache.Remove(slot.Id, numKeep, numKeep+discard)
 		if err != nil {
-			return fmt.Errorf("unable to remove old kv cache entries (id: %v, keep: %v discard: %v): %w", slot.Id, numKeep, discard, err)
+			slog.Debug("kv cache removal failed, clearing cache and returning inputs for reprocessing",
+				"id", slot.Id, "error", err)
+
+			// Clear the entire KV cache
+			_ = c.cache.Remove(slot.Id, 0, -1)
+
+			// Create new input slice with preserved tokens (numKeep + remaining tokens after discard)
+			newInputs := make([]input, numKeep+inputLen-(numKeep+discard))
+			copy(newInputs[:numKeep], slot.Inputs[:numKeep])
+			copy(newInputs[numKeep:], slot.Inputs[numKeep+discard:])
+
+			// Reset the slot inputs since we've cleared the cache
+			slot.Inputs = []input{}
```

Comment:
Minor but maybe move this line up so that it follows `c.cache.Remove`?

Reply:
I think the current order is actually correct here, as we need to copy data from slot.Inputs before clearing it.

Reply:
Good point but maybe in that case we can move `c.cache.Remove(slot.Id, 0, -1)` down? Just trying to make sure these two things stay in sync.

Reply:
Moved the cache removal down